src/lib/__tests__/index-test.ts

```diff
@@ -0,0 +1,855 @@
+
+      instance.init!(
+        createInitOptions({
+          instantSearchInstance,
+        })
+      );
+
+      // Simulate a call to search from a widget
+      instance.getHelper()!.searchForFacetValues('brand', 'Apple', 10, {
+        highlightTags: 'mark',
+      });
+
+      expect(searchClient.searchForFacetValues).toHaveBeenCalledTimes(1);
+      expect(searchClient.searchForFacetValues).toHaveBeenCalledWith(
+        expect.arrayContaining([
+          {
+            indexName: 'index_name',
+            params: expect.objectContaining({
+              facetName: 'brand',
+              facetQuery: 'Apple',
+              maxFacetHits: 10,
+              highlightTags: 'mark',
+              page: 5,
+            }),
+          },
+        ])
+      );
+    });
+
+    it('schedules a render on DerivedHelper results', async () => {
+      const instance = index({ indexName: 'index_name' });
+      const searchClient = createSearchClient();
+      const mainHelper = algoliasearchHelper(searchClient, '', {});
+      const instantSearchInstance = createInstantSearch({
+        mainHelper,
+      });
+
+      instance.init!(
+        createInitOptions({
+          instantSearchInstance,
+        })
+      );
+
+      // Simulate a call to search from a widget
+      instance.getHelper()!.search();
+
+      expect(instantSearchInstance.scheduleRender).toHaveBeenCalledTimes(0);
+
+      await runAllMicroTasks();
+
+      expect(instantSearchInstance.scheduleRender).toHaveBeenCalledTimes(1);
+    });
+
+    it('schedules a stalled render on DerivedHelper search', () => {
+      const instance = index({ indexName: 'index_name' });
+      const searchClient = createSearchClient();
+      const mainHelper = algoliasearchHelper(searchClient, '', {});
+      const instantSearchInstance = createInstantSearch({
+        mainHelper,
+      });
+
+      instance.init!(
+        createInitOptions({
+          instantSearchInstance,
+        })
+      );
+
+      expect(instantSearchInstance.scheduleStalledRender).toHaveBeenCalledTimes(
+        0
+      );
+
+      // Simulate a call to search from a widget
+      instance.getHelper()!.search();
+
+      expect(instantSearchInstance.scheduleStalledRender).toHaveBeenCalledTimes(
+        1
+      );
+    });
+
+    it('calls `init` on its widgets', () => {
+      const instance = index({ indexName: 'index_name' });
+      const instantSearchInstance = createInstantSearch();
+
+      const searchBox = createSearchBox({
+        init: jest.fn(),
+      });
+
+      const pagination = createSearchBox({
+        init: jest.fn(),
+      });
+
+      instance.addWidgets([searchBox, pagination]);
+
+      expect(searchBox.init).toHaveBeenCalledTimes(0);
+      expect(pagination.init).toHaveBeenCalledTimes(0);
+
+      instance.init!(
+        createInitOptions({
+          instantSearchInstance,
+        })
+      );
+
+      expect(searchBox.init).toHaveBeenCalledTimes(1);
```

Comment:
We don't test what `searchBox.init` was called with?

Reply:
bd2b7f77ad8405f4b862d0929064896905ef03ad